ooiservices/app/uframe/data.py

```diff
@@ -84,24 +84,27 @@ def get_data(stream, instrument, yfields, xfields, include_time=True):
     # xdata = OrderedDict()
     # ydata = OrderedDict()
 
-    x = OrderedDict({k: np.empty(len(data)) for k in xfields})
-    y = OrderedDict({k: np.empty(len(data)) for k in yfields})
+    x = OrderedDict({k: np.empty(len(data)-1) for k in xfields})
+    y = OrderedDict({k: np.empty(len(data)-1) for k in yfields})    
 
     for ind, row in enumerate(data):
-        # used to handle multiple streams
-        if row['stream_name'] == stream:
-            # x
-            for xfield in xfields:
-                if xfield == 'time':
-                    x[xfield][ind] = float(row['pk']['time'])
-                else:
-                    x[xfield][ind] = row[xfield]
-            # y
-            for yfield in yfields:
-                if yfield == 'time':
-                    y[yfield][ind] = float(row['pk']['time'])
-                else:
-                    y[yfield][ind] = row[yfield]
+        #skip the last row
+        if ind < len(data)-1:
+            # used to handle multiple streams
+            if row['pk']['stream'] == stream:
```

Comment:
@Bobfrat you will need to add this, as we check the pk stream, which didnt exist in the old version of uframe services